packages/camera/camera_avfoundation/ios/camera_avfoundation/Sources/camera_avfoundation/CameraPlugin.swift

```diff
@@ -0,0 +1,528 @@
+          lensFacing = .front
+        case .unspecified:
+          lensFacing = .external
+        @unknown default:
+          lensFacing = .external
+        }
+
+        let cameraDescription = FCPPlatformCameraDescription.make(
+          withName: device.uniqueID,
+          lensDirection: lensFacing
+        )
+        reply.append(cameraDescription)
+      }
+
+      completion(reply, nil)
+    }
+  }
+
+  public func createCamera(
+    withName cameraName: String,
+    settings: FCPPlatformMediaSettings,
+    completion: @escaping (NSNumber?, FlutterError?) -> Void
+  ) {
+    // Create FLTCam only if granted camera access (and audio access if audio is enabled)
+    captureSessionQueue.async { [weak self] in
+      self?.permissionManager.requestCameraPermission { error in
+        guard let strongSelf = self else { return }
+
+        if let error = error {
+          completion(nil, error)
+          return
+        }
+
+        // Request audio permission on `create` call with `enableAudio` argument instead of the
+        // `prepareForVideoRecording` call. This is because `prepareForVideoRecording` call is
+        // optional, and used as a workaround to fix a missing frame issue on iOS.
+        if settings.enableAudio {
+          // Setup audio capture session only if granted audio access.
+          strongSelf.permissionManager.requestAudioPermission { [weak self] audioError in
+            // cannot use the outter `strongSelf`
+            guard let strongSelf = self else { return }
+
+            if let audioError = audioError {
+              completion(nil, audioError)
+              return
+            }
+
+            strongSelf.createCameraOnSessionQueue(
+              withName: cameraName,
+              settings: settings,
+              completion: completion)
+          }
+        } else {
+          strongSelf.createCameraOnSessionQueue(
+            withName: cameraName,
+            settings: settings,
+            completion: completion)
+        }
+      }
+    }
+  }
+
+  public func createCameraOnSessionQueue(
+    withName: String,
+    settings: FCPPlatformMediaSettings,
+    completion: @escaping (NSNumber?, FlutterError?) -> Void
+  ) {
+    captureSessionQueue.async { [weak self] in
+      self?.sessionQueueCreateCamera(name: withName, settings: settings, completion: completion)
+    }
+  }
+
+  // This must be called on captureSessionQueue. It is extracted from
+  // initializeCamera:withImageFormat:completion: to make it easier to reason about strong/weak
+  // self pointers.
+  private func sessionQueueCreateCamera(
+    name: String,
+    settings: FCPPlatformMediaSettings,
+    completion: @escaping (NSNumber?, FlutterError?) -> Void
+  ) {
+    let mediaSettingsAVWrapper = FLTCamMediaSettingsAVWrapper()
+
+    let camConfiguration = FLTCamConfiguration(
+      mediaSettings: settings,
+      mediaSettingsWrapper: mediaSettingsAVWrapper,
+      captureDeviceFactory: { [self] in
```

Comment:
weak self

Reply:
is it updated or am i reading an old version?

Reply:
I haven't updated this yet. I think I will propagate the "named" factory here. It will
1. Remove the lambda completely thus resolving the strong/weak self issue completely without requiring additional handling for nil self
2. Remove the possible "named"/"unnamed" factory confusion
3. Mostly resolve this [issue](https://github.com/flutter/flutter/issues/162376) except for the error handling/logic issues in the `setDescriptionWhileRecording` method

Reply:
I've updated the code and the PR description to match